src/Microsoft.ApiDesignGuidelines.Analyzers/Core/TypeNamesShouldNotMatchNamespaces.cs

```diff
@@ -25,24 +29,236 @@ public abstract class TypeNamesShouldNotMatchNamespacesAnalyzer : DiagnosticAnal
                                                                              s_localizableMessageDefault,
                                                                              DiagnosticCategory.Naming,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182257.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
         internal static DiagnosticDescriptor SystemRule = new DiagnosticDescriptor(RuleId,
                                                                              s_localizableTitle,
                                                                              s_localizableMessageSystem,
                                                                              DiagnosticCategory.Naming,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182257.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(DefaultRule, SystemRule);
 
+        private static object s_obj = new object();
+        private static ImmutableDictionary<string, string> s_wellKnownSystemNamespaceTable;
+
         public override void Initialize(AnalysisContext analysisContext)
         {
+            analysisContext.RegisterCompilationStartAction(
+                compilationStartAnalysisContext =>
+                {
+                    var namedTypesInCompilation = new ConcurrentBag<INamedTypeSymbol>();
```

Comment:
Normally individual actions within an analyzer don't run in parallel and so you don't need a ConcurrentBag. However you can ask for these actions to be run in parallel for better perf by calling `analysisContext.EnableConcurrentExecution()`. If you think this analyzer is thread-safe then please add that to the beginning of the Initialize method.